### PR TITLE
hotfix: prevent exception in record_one_broadcast_with_new_relic

### PR DESCRIPTION
**Pull request by nathan-appere**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Describe the changes present in the pull request

Submitter Checklist:
- [ ] Include a link to the related GitHub issue, if applicable
- [ ] Include a security review link, if applicable

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
GitHub Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

File: lib/new_relic/agent/instrumentation/active_support_broadcast_logger/instrumentation.rb [hunk 1]
```diff
@@ -5,9 +5,13 @@
...
     def record_one_broadcast_with_new_relic(*args)
-      broadcasts[1..-1].each { |broadcasted_logger| broadcasted_logger.instance_variable_set(:@skip_instrumenting, true) }
-      yield
-      broadcasts.each { |broadcasted_logger| broadcasted_logger.instance_variable_set(:@skip_instrumenting, false) }
+      if broadcasts && broadcasts[1..-1]
+        broadcasts[1..-1].each { |broadcasted_logger| broadcasted_logger.instance_variable_set(:@skip_instrumenting, true) }
+        yield
+        broadcasts.each { |broadcasted_logger| broadcasted_logger.instance_variable_set(:@skip_instrumenting, false) }
+      else
+        ::NewRelic::Agent.record_custom_event('NR Error - record_one_broadcast_with_new_relic - nil broadcasts', {})
```
Comment (hannahramadan) [inline lib/new_relic/agent/instrumentation/active_support_broadcast_logger/instrumentation.rb:13]: Is there a something you hoped to get out of using `record_custom_event`?

We'd typically log an error instead, something like:

```suggestion
        NewRelic::Agent.logger.error('Error recording broadcasted logger')
```

**Comment by hannahramadan** [comment]
Hi @nathan-appere! Thank you so much for this contribution. I've left one small comment and we will need the [Contributor License Agreement](https://cla-assistant.io/newrelic/newrelic-ruby-agent?pullRequest=2510) signed before merging ◡̈

**Comment by nathan-appere** [comment]
Hey @hannahramadan thanks for having a look, I ended up not deploying as it seems a bit dangerous to have a Rails version with random errors 🤔

**Review by tannalynn — APPROVED** [review]
Thank you for getting this updated and signing the CLA!

**Pull request merged**